cirq/ops/two_qubit_diagonal_gate.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common quantum gates that target three qubits."""
+
+from typing import Any, List, TYPE_CHECKING
+
+import numpy as np
+import sympy
+
+from cirq import protocols, value
+from cirq._compat import proper_repr
+from cirq.ops import gate_features
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import
+    import cirq
+
+
+@value.value_equality()
+class TwoQubitDiagonalGate(gate_features.TwoQubitGate):
+    """A gate given by a diagonal 4x4 matrix."""
+
+    def __init__(self, diag_angles_radians: List[value.TParamVal]) -> None:
+        r"""A two qubit gate with only diagonal elements.
+
+        This gate's off-diagonal elements are zero and it's on diagonal
+        elements are all phases.
+
+        Args:
+            diag_angles_radians: The list of angles on the diagonal in radians.
+                If these values are $(x_0, x_1, \ldots , x_3)$ then the unitary
+                has diagonal values $(e^{i x_0}, e^{i x_1}, \ldots, e^{i x_3})$.
+        """
+        self._diag_angles_radians: List[value.TParamVal] = diag_angles_radians
```

Comment:
It may be a good idea to copy the argument (`= np.array(diag_angle_radians, dtype=np.float64)` or `= tuple(diag_angle_radians)`) to avoid accidental gate mutation.